apps/www/components/Audio/AudioPlayerContainer.tsx

```diff
@@ -0,0 +1,622 @@
+      trackEvent([AUDIO_PLAYER_TRACK_CATEGORY, 'error', error])
+    } else {
+      reportError(
+        'handle audio-error',
+        JSON.stringify(error, Object.getOwnPropertyNames(error), 2),
+      )
+      trackEvent([
+        AUDIO_PLAYER_TRACK_CATEGORY,
+        'error',
+        JSON.stringify(error, Object.getOwnPropertyNames(error), 2),
+      ])
+    }
+  }
+
+  const saveActiveItemProgress = useCallback(
+    async (forcedState?: { currentTime?: number; isPlaying?: boolean }) => {
+      const { mediaId } = activePlayerItem?.document.meta?.audioSource ?? {}
+      if (duration < (forcedState?.currentTime ?? currentTime)) {
+        trackEvent([
+          AUDIO_PLAYER_TRACK_CATEGORY,
+          'updateProgress',
+          'illegalUpdate',
+          `upsert-value was ${
+            forcedState?.currentTime ?? currentTime
+          }, track duration is ${duration}, mediaId is ${mediaId}`,
+        ])
+        return
+      }
+
+      return saveMediaProgress(
+        mediaId,
+        forcedState?.currentTime ?? currentTime,
+        forcedState?.isPlaying ?? isPlaying,
+      )
+    },
+    [activePlayerItem, currentTime, isPlaying, saveMediaProgress, duration],
+  )
+
+  const syncWithNativeApp = (state: AudioPlayerState) => {
+    if (!inNativeApp) return
+    // Don't update the currentTime & duration
+    // if the native player wasn't yet able to load the media
+    // TODO: find out why the track-player sometimes provides currentTime = 0
+    // when state is ready
+    if (
+      ![
+        NativeAudioPlayerState.None,
+        NativeAudioPlayerState.Connecting,
+      ].includes(state.playerState)
+    ) {
+      setDuration(state.duration)
+      setCurrentTime(state.currentTime)
+    }
+    setPlaybackRate(state.playRate)
+    setIsPlaying(
+      state?.isPlaying ||
+        [NativeAudioPlayerState.Playing, NativeAudioPlayerState.Ready].includes(
+          state.playerState,
+        ),
+    )
+    setIsLoading(
+      state?.isLoading ||
+        [
+          NativeAudioPlayerState.Buffering,
+          NativeAudioPlayerState.Connecting,
+          NativeAudioPlayerState.None,
+        ].includes(state.playerState),
+    )
+  }
+
+  const syncWithMediaElement = () => {
+    try {
+      if (!mediaRef.current) return
+      const audioElem = mediaRef.current
+      setCurrentTime(audioElem.currentTime)
+      setDuration(audioElem.duration)
+      setPlaybackRate(audioElem.playbackRate)
+      setIsPlaying(!audioElem.paused)
+      setIsLoading(audioElem.readyState < 2)
+      setBuffered(audioElem.buffered)
+    } catch (error) {
+      handleError(error)
+    }
+  }
+
+  const onCanPlay = async () => {
+    try {
+      setIsLoading(false)
+      syncWithMediaElement()
+
+      if (activePlayerItem?.id !== trackedPlayerItem?.current?.id) {
+        trackedPlayerItem.current = activePlayerItem
+
+        const { userProgress, durationMs } =
+          activePlayerItem.document?.meta.audioSource ?? {}
```

Comment:
Use LoadMediaProgress instead to also make use of progress stored in the local-storage.